components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -780,60 +780,93 @@ public IdentityProvider getIdentityProvider(String federatedIdPName, String tena
     public IdentityProvider[] getAllIdentityProviders(String tenantDomain)
             throws IdentityApplicationManagementException {
 
+        List<IdentityProvider> fedIdpList;
         try {
             startTenantFlow(tenantDomain);
             IdentityProviderDAO idpdao = ApplicationMgtSystemConfig.getInstance().getIdentityProviderDAO();
-            List<IdentityProvider> fedIdpList = idpdao.getAllIdentityProviders();
-            if (fedIdpList != null) {
-                return fedIdpList.toArray(new IdentityProvider[fedIdpList.size()]);
-            }
-            return new IdentityProvider[0];
+            fedIdpList = idpdao.getAllIdentityProviders();
         } catch (Exception e) {
             String error = "Error occurred while retrieving all Identity Providers" + ". " + e.getMessage();
             throw new IdentityApplicationManagementException(error, e);
         } finally {
             endTenantFlow();
         }
+
+        // Invoking the listeners
+        Collection<ApplicationMgtListener> listeners =
```

Comment:
Don't we need to check fedIdp nullability before invoking listener doPostGetAllIdentityProviders? Can we pass a null value to this method?